lib/improver/nbhood/use_nbhood.py

```diff
@@ -156,3 +159,133 @@ def process(self, cube, mask_cube):
             cube, concatenated_cube,
             exception_coordinates=exception_coordinates)
         return concatenated_cube
+
+
+class CollapseMaskedNeighbourhoodCoordinate(object):
+
+    """
+    Plugin for collapsing the coordinate the mask was applied to after
+    masked neighbourhood processing.
+
+    Takes into account the result from the neighbourhood processing to
+    adjust the weights between the bands in the coordinate for the points
+    where the were no points within a neighbourhood for and a non-zero
+    weighting.
```

Comment:
This is quite general, but there is a very specific application envisaged (ie collapsing over topographic bands after neighbourhooding).  Could this be added as a usage example to the docstring?

Reply:
I've added this, but let me know what you think of it.